### DISCUSSION TITLE
v2.0.20

### DISCUSSION THREAD

**Discussion by DTTerastar** [ESPresense/ESPresense#259] [Announcements]
## Changes

- Add include exclude query filtering @DTTerastar (#255)

<hr /><em>This discussion was created from the release <a href='https://github.com/ESPresense/ESPresense/releases/tag/v2.0.20'>v2.0.20</a>.</em>

**Comment by DTTerastar** [comment]
This is a BIG release.  Querying has been completely revamped to take over the ESP less.  The ESP cannot listen for advertisements while querying so it's important to be as judicious with your use of them.  They are only useful if a certain apple:10* fingerprint isn't unique enough, and it'll only help if the 2 devices are different models.

**Comment by k3vyn32** [comment]
This release does not appear to be picking up my roomAssistant IDs any more.
Just see loads of apple: IDs, and a couple of iBeacon ones.
The roomAssistant is definitely still there and working as I can see it with Bluetility, and can connect to a Pi with Room Assistant still running, if I open up the Max Distance.
Any thoughts?

**Reply by Schlurcher** [reply]
Same for me, Bluetility picks it up fine. Nothing on MQTT. Gonna plug into it tomorrow and have a look at the console.

**Reply by Schlurcher** [reply]
Okay, for some reason it gets reported as the name of another id?

<img width="1293" alt="CleanShot 2022-02-14 at 02 20 00@2x" src="https://user-images.githubusercontent.com/973681/153784888-fc141d15-8af2-457d-825b-f5bee22a2f68.png">

**Reply by DTTerastar** [reply]
Yes, i seem to have broken all querying.  I think it's due to trying to downgrade NimBLE.  Going to revert that now.

**Reply by Schlurcher** [reply]
Still no dice with 2.0.21. I've plugged into the console and installed the verbose flavor. roomAssistant isn't showing up on the console. With Bluetility on a Mac it shows up.

**Reply by k3vyn32** [reply]
Same for me also I am afraid.

**Reply by k3vyn32** [reply]
All good again - thanks
